compiler/gen_metadata.ts

```diff
@@ -2,6 +2,93 @@ import * as ts from "typescript"
 import * as fs from "fs"
 import * as commandLineArgs from "command-line-args"
 
+class SimpleType {
+    name: string
+    kind: string
+    // Support for generics is mostly done but it might not be needed for now
+    // generics: Array<SimpleType>
+    arguments: Array<SimpleType>
+    type: SimpleType
+
+    constructor(typeNode) {
+        switch (typeNode.kind) {
+            case 105:
```

Comment:
Could you link to the documentation where these magic numbers are specified, in a comment? It seems like the kind of thing that's so deep within their API that they might just change it without saying anything.

Reply:
I feel extremely dumb. What I did to find the information I needed was that I ran the compiler in a debugger and basically put breakpoints where interesting things might happen. One of these places was a switch/case where all these numbers appeared so I just copied what I saw. The thought that I might be staring at TypeScript enums that had been translated to numbers didn't cross my mind at all and now that I took a look at the TypeScript source code I can see that these values have proper names. I'll update my PR to fix that.

I couldn't find any documentation about the TypeScript API though. Compilers seem to have little to no documentation besides what you can find in the source code.